odata_query/roundtrip.py

```diff
@@ -0,0 +1,195 @@
+import sys
+from typing import Type
+
+if sys.version_info >= (3, 8):
```

Comment:
Wouldn't `from typing_extensions import Protocol` work for both cases?

Reply:
It would, but `typing_extensions` is only installed on lower Python versions:
```
typing-extensions = {version = "*", markers = "python_version < \"3.8\""}
```